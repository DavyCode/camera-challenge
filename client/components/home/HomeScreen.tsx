import { Fragment } from 'react';
import Button from '../ui/button';
import classes from './HomeScreen.module.css';

function HomeScreen() {
  return (
    <Fragment>
      <main className={classes.main}>
        <h1 className={classes.title}>
          <p>Amazing Camera App</p>
        </h1>
        <div data-testid="home-button">
          <Button link="/capture">
            <span> Start Capture </span>
          </Button>
        </div>
      </main>
    </Fragment>
  );
}

export default HomeScreen;
