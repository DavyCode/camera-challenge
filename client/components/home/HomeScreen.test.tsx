import { render, fireEvent, screen, cleanup } from '@testing-library/react';
import { useRouter } from 'next/router';
import HomeScreen from './HomeScreen';

jest.mock('next/router', () => ({
  useRouter() {
    return {
      route: '/capture',
    };
  },
}));

afterEach(cleanup);

describe('HomeScreen', () => {
  const renderComponent = () => render(<HomeScreen />);

  it('Should render', () => {
    render(<HomeScreen />);
    const main = screen.getByRole('main');
    expect(main).toBeInTheDocument();
  });

  it('Should match snapshot', () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  });

  it('Should navigate to Capture /capture page', () => {
    const { getByTestId } = render(<HomeScreen />);

    fireEvent.click(getByTestId('home-button'));

    expect(useRouter().route).toMatch('/capture');
  });
});
