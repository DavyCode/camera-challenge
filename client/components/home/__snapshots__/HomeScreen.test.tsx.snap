// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen Should match snapshot 1`] = `
<div>
  <main
    class="main"
  >
    <h1
      class="title"
    >
      <p>
        Amazing Camera App
      </p>
    </h1>
    <div
      data-testid="home-button"
    >
      <a
        class="btn"
        href="/capture"
      >
        <span>
           Start Capture 
        </span>
      </a>
    </div>
  </main>
</div>
`;
